Source/ASCollectionView.mm

```diff
@@ -576,6 +576,11 @@ - (void)setAsyncDelegate:(id<ASCollectionDelegate>)asyncDelegate
 - (void)_asyncDelegateOrDataSourceDidChange
 {
   ASDisplayNodeAssertMainThread();
+  
+  if (ASActivateExperimentalFeature(ASExperimentalSkipClearData)) {
```

Comment:
Checking for `nil`s is less expensive than checking for experiment, so let's do it first, e.g:

```
  if (_asyncDataSource == nil && _asyncDelegate == nil && ASActivateExperimentalFeature(ASExperimentalSkipClearData)) {
    [_dataController clearData];
  }
```

Reply:
Good point